### ISSUE TITLE
Unable to run interpreter after building

### ISSUE THREAD

**Issue by Vici37** [crystal-lang/crystal#14331]
## Bug Report

I'm experimenting with the crystal interpreter and learning the inner workings of it. So far this has been through the [ic](https://github.com/I3oris/ic) shard, as I like its repl support more. I recently found that `require "http/server"` is running into a bug in the interpreter but not the compiler, and wanted to double check if that's a known issue and / or reproducible with crystal's own interpreter.

Running:
```
> git pull
> make clean
> make interpreter=1
> ./.build/crystal i
```

Yields the output:
```
Crystal interpreter 1.12.0-dev [e3eb178f2] (2024-02-27).
EXPERIMENTAL SOFTWARE: if you find a bug, please consider opening an issue in
https://github.com/crystal-lang/crystal/issues/new/
Error: 
```
No additional error details are provided. `Error:` is the end of the output :/ I see https://github.com/crystal-lang/crystal/issues/14195 as the most recent open bug for the interpreter, but couldn't figure out if this is the same bug.

Running on Linux Mint 20.3, LLVM 12 installed, `make interpreter_spec` passes. Thanks for making crystal!

**Comment by Blacksmoke16** [comment]
Try using `./bin/crystal i` instead of `.build/crystal i`

**Comment by Vici37** [comment]
Well now I feel silly :) I could have sworn the last time I tried that, it reported it was still using `1.10.0-dev` (the last time I built, I guess) afterwards, and figured  `.build/crystal` was correct. Now I'm seeing `1.12.0-dev`. Thanks @Blacksmoke16 !

I'm now able to verify the same bug from running `require "http/server"` here too, will create a separate issue for it.

(error is `Error: method HTTP::Server::Response::Output#sync? must return Bool but it is returning (Bool | Nil)`)
